streatham_go/app/views/leaderboard.py

```diff
@@ -1,7 +1,48 @@
 from django.shortcuts import render
 from django.contrib.auth.decorators import login_required
+from ..models import Leaderboard
+
+
+def get_current_user_rank(user, user_data, current_user_data):
+    return user_data.index(
+        {
+            'username': user.username,
+            'level': current_user_data.level,
+            'quiz_count': current_user_data.quiz_count
+        }) + 1
+
+
+def get_current_user_data(user, leaderboard_data):
+    return leaderboard_data.filter(user=user).first()
+
+
+def get_user_data(leaderboard_data):
+    user_data = []
+    for user_leaderboard_data in leaderboard_data:
+        user_data.append({
+            'username': user_leaderboard_data.user.username,
+            'level': user_leaderboard_data.level,
+            'quiz_count': user_leaderboard_data.quiz_count
+        })
+    return user_data
```

Comment:
Move to leaderboard class as class methods

Reply:
Done